From f3df2254a8d468226e332d61b168da5eba232d1d Mon Sep 17 00:00:00 2001
From: Alexander Martinz <amartinz@shiftphones.com>
Date: Wed, 25 Jan 2023 12:36:58 +0100
Subject: [PATCH 2/3] media: kona: Drop OMX sw video codecs

These are already supported by c2.

3gpp:
  - c2.android.h263.decoder
  - c2.android.h263.encoder

mp4v-es:
  - c2.android.mpeg4.decoder
  - c2.android.mpeg4.encoder

Test: run cts -m CtsMediaDecoderTestCases
Test: run cts -m CtsMediaEncoderTestCases

Change-Id: I376311d32621ad852a2a885a51132737d94a767e
Signed-off-by: Alexander Martinz <amartinz@shiftphones.com>
---
 conf_files/kona/media_codecs.xml             | 69 --------------------
 conf_files/kona/media_codecs_performance.xml | 19 ------
 conf_files/kona/media_codecs_vendor.xml      | 69 --------------------
 3 files changed, 157 deletions(-)

diff --git a/conf_files/kona/media_codecs.xml b/conf_files/kona/media_codecs.xml
index eb60781..534174b 100644
--- a/conf_files/kona/media_codecs.xml
+++ b/conf_files/kona/media_codecs.xml
@@ -118,8 +118,6 @@ Only the three quirks included above are recognized at this point:
  |             | 3820    2160    240     220     7735500             |
  |             | 4096    2160    120     220     4147200             |
  |             | 7680    4320    60      220     7776000             |
- | mpeg4-sw    | 1920    1088    30      40      244800      244800  |
- |             | 1280    720     30      40      108000              |
  | vp8         | 1280    720     120     220     432000      1036800 |
  |             | 1920    1088    120     220     979200              |
  |             | 3840    2160    30      220     972000              |
@@ -128,7 +126,6 @@ Only the three quirks included above are recognized at this point:
  |             | 3820    2160    240     220     7735500             |
  |             | 4096    2160    120     220     4147200             |
  |             | 7680    4320    60      220     7776000             |
- | h263-sw     | 864     480     30      16      48600       48600   |
  | mpeg2       | 1020    1088    60i     40      130050      244800  |
  |             | 1920    1088    30      40      244800              |
  |_____________|_____________________________________________________|
@@ -172,12 +169,10 @@ Only the three quirks included above are recognized at this point:
  |          | 4096    2160    60      220     2073600             |
  |          | 8192    4320    24      220     3317760             |
  | hevc.cq  | 512     512     3825    220     3916800     3916800 |
- | mpeg4-sw | 1280    720     30      4       108000      108000  |
  | vp8      | 1280    720     60      70      216000      2073600 |
  |          | 1920    1088    60      70      489600              |
  |          | 3840    2160    60      70      1944000             |
  |          | 4096    2160    60      70      2073600             |
- | h263-sw  | 864     480     30      2       48600       48600   |
  |__________|_____________________________________________________|
 -->
 
@@ -268,40 +263,6 @@ Only the three quirks included above are recognized at this point:
             <Limit name="performance-point-16384x8192" value="3" />
             <Limit name="performance-point-1920x1080" value="6" />
         </MediaCodec>
-        <!-- Video Software -->
-        <MediaCodec name="OMX.qcom.video.encoder.h263sw" type="video/3gpp" >
-            <Quirk name="requires-allocate-on-input-ports" />
-            <Quirk name="requires-allocate-on-output-ports" />
-            <Quirk name="requires-loaded-to-idle-after-allocation" />
-            <Limit name="size" min="96x96" max="864x864" />
-            <Limit name="alignment" value="4x4" />
-            <Limit name="block-size" value="16x16" />
-            <Limit name="block-count" range="36-1620" />
-            <Limit name="blocks-per-second" min="36" max="48600" />
-            <Limit name="bitrate" range="1-2000000" />
-            <Limit name="frame-rate" range="1-30" />
-            <Limit name="concurrent-instances" max="3" />
-            <Limit name="performance-point-720x576" value="25" />
-            <Limit name="performance-point-720x480" value="30" />
-            <Limit name="performance-point-864x480" value="30" />
-        </MediaCodec>
-        <MediaCodec name="OMX.qcom.video.encoder.mpeg4sw" type="video/mp4v-es" >
-             <Quirk name="requires-allocate-on-input-ports" />
-             <Quirk name="requires-allocate-on-output-ports" />
-             <Quirk name="requires-loaded-to-idle-after-allocation" />
-             <Limit name="size" min="96x96" max="864x864" />
-             <Limit name="alignment" value="2x2" />
-             <Limit name="block-size" value="16x16" />
-             <Limit name="block-count" range="36-1620" />
-             <Limit name="blocks-per-second" min="36" max="48600" />
-             <Limit name="bitrate" range="1-8000000" />
-             <Limit name="frame-rate" range="1-30" />
-             <Attribute name="software-codec" />
-             <Limit name="concurrent-instances" max="3" />
-             <Limit name="performance-point-720x576" value="25" />
-             <Limit name="performance-point-720x480" value="30" />
-             <Limit name="performance-point-864x480" value="30" />
-        </MediaCodec>
     </Encoders>
     <Decoders>
        <!-- Video Hardware  -->
@@ -455,36 +416,6 @@ Only the three quirks included above are recognized at this point:
             <Limit name="performance-point-3840x2160" value="60" />
             <Limit name="performance-point-4096x2304" value="60" />
         </MediaCodec>
-        <!-- Video Software -->
-        <MediaCodec name="OMX.qti.video.decoder.h263sw" type="video/3gpp" >
-             <Quirk name="requires-allocate-on-input-ports" />
-             <Quirk name="requires-allocate-on-output-ports" />
-             <Limit name="size" min="96x96" max="864x864" />
-             <Limit name="alignment" value="4x4" />
-             <Limit name="block-size" value="16x16" />
-             <Limit name="block-count" range="36-1620" />
-             <Limit name="blocks-per-second" min="36" max="48600" />
-             <Limit name="bitrate" range="1-16000000" />
-             <Limit name="frame-rate" range="1-30" />
-             <Feature name="adaptive-playback" />
-             <Limit name="concurrent-instances" max="4" />
-             <Limit name="performance-point-720x480" value="30" />
-             <Limit name="performance-point-720x576" value="25" />
-             <Limit name="performance-point-864x480" value="30" />
-        </MediaCodec>
-        <MediaCodec name="OMX.qti.video.decoder.mpeg4sw" type="video/mp4v-es">
-             <Quirk name="requires-allocate-on-input-ports" />
-             <Quirk name="requires-allocate-on-output-ports" />
-             <Limit name="size" min="96x96" max="1920x1920" />
-             <Limit name="alignment" value="2x2" />
-             <Limit name="block-size" value="16x16" />
-             <Limit name="block-count" range="36-8160" />
-             <Limit name="blocks-per-second" min="36" max="244800" />
-             <Limit name="bitrate" range="1-40000000" />
-             <Limit name="frame-rate" range="1-30" />
-             <Limit name="concurrent-instances" max="4" />
-             <Limit name="performance-point-1920x1080" value="30" />
-        </MediaCodec>
         <MediaCodec name="OMX.google.opus.decoder" type="audio/opus" update="true" rank="100"/>
     </Decoders>
 </MediaCodecs>
diff --git a/conf_files/kona/media_codecs_performance.xml b/conf_files/kona/media_codecs_performance.xml
index 6fc6cdb..7316612 100644
--- a/conf_files/kona/media_codecs_performance.xml
+++ b/conf_files/kona/media_codecs_performance.xml
@@ -37,16 +37,6 @@ limitations under the License.
         <MediaCodec name="OMX.qcom.video.encoder.hevc.cq" type="video/hevc" update="true">
             <Limit name="measured-frame-rate-320x240" range="340-400" />
         </MediaCodec>
-        <MediaCodec name="OMX.qcom.video.encoder.h263sw" type="video/3gpp" update="true">
-            <Limit name="measured-frame-rate-176x144" range="303-303" />
-            <Limit name="measured-frame-rate-352x288" range="259-259" />
-            <Limit name="measured-frame-rate-704x576" range="87-155" />
-        </MediaCodec>
-        <MediaCodec name="OMX.qcom.video.encoder.mpeg4sw" type="video/mp4v-es" update="true">
-            <Limit name="measured-frame-rate-176x144" range="299-299" />
-            <Limit name="measured-frame-rate-352x288" range="120-170" />
-            <Limit name="measured-frame-rate-640x480" range="148-148" />
-        </MediaCodec>
         <MediaCodec name="OMX.qcom.video.encoder.vp8" type="video/x-vnd.on2.vp8" update="true">
             <Limit name="measured-frame-rate-320x180" range="355-520" />
             <Limit name="measured-frame-rate-640x360" range="236-302" />
@@ -101,15 +91,6 @@ limitations under the License.
             <Limit name="measured-frame-rate-1920x1080" range="316-318" />
             <Limit name="measured-frame-rate-3840x2160" range="148-149" />
         </MediaCodec>
-        <MediaCodec name="OMX.qti.video.decoder.h263sw" type="video/3gpp" update="true">
-            <Limit name="measured-frame-rate-176x144" range="782-1581" />
-            <Limit name="measured-frame-rate-352x288" range="380-920" />
-        </MediaCodec>
-        <MediaCodec name="OMX.qti.video.decoder.mpeg4sw" type="video/mp4v-es" update="true">
-            <Limit name="measured-frame-rate-176x144" range="775-2064" />
-            <Limit name="measured-frame-rate-480x360" range="695-915" />
-            <Limit name="measured-frame-rate-1280x720" range="369-529" />
-        </MediaCodec>
         <MediaCodec name="OMX.qcom.video.decoder.vp8" type="video/x-vnd.on2.vp8" update="true">
             <Limit name="measured-frame-rate-320x240" range="1111-1357" />
             <Limit name="measured-frame-rate-640x360" range="571-883" />
diff --git a/conf_files/kona/media_codecs_vendor.xml b/conf_files/kona/media_codecs_vendor.xml
index 62b0b66..b2bbb7c 100644
--- a/conf_files/kona/media_codecs_vendor.xml
+++ b/conf_files/kona/media_codecs_vendor.xml
@@ -118,8 +118,6 @@ Only the three quirks included above are recognized at this point:
  |             | 3820    2160    240     220     7735500             |
  |             | 4096    2160    120     220     4147200             |
  |             | 7680    4320    60      220     7776000             |
- | mpeg4-sw    | 1920    1088    30      40      244800      244800  |
- |             | 1280    720     30      40      108000              |
  | vp8         | 1280    720     120     220     432000      1036800 |
  |             | 1920    1088    120     220     979200              |
  |             | 3840    2160    30      220     972000              |
@@ -128,7 +126,6 @@ Only the three quirks included above are recognized at this point:
  |             | 3820    2160    240     220     7735500             |
  |             | 4096    2160    120     220     4147200             |
  |             | 7680    4320    60      220     7776000             |
- | h263-sw     | 864     480     30      16      48600       48600   |
  | mpeg2       | 1020    1088    60i     40      130050      244800  |
  |             | 1920    1088    30      40      244800              |
  |_____________|_____________________________________________________|
@@ -172,12 +169,10 @@ Only the three quirks included above are recognized at this point:
  |          | 4096    2160    60      220     2073600             |
  |          | 8192    4320    24      220     3317760             |
  | hevc.cq  | 512     512     3825    220     3916800     3916800 |
- | mpeg4-sw | 1280    720     30      4       108000      108000  |
  | vp8      | 1280    720     60      70      216000      2073600 |
  |          | 1920    1088    60      70      489600              |
  |          | 3840    2160    60      70      1944000             |
  |          | 4096    2160    60      70      2073600             |
- | h263-sw  | 864     480     30      2       48600       48600   |
  |__________|_____________________________________________________|
 -->
 
@@ -269,40 +264,6 @@ Only the three quirks included above are recognized at this point:
             <Limit name="performance-point-16384x8192" value="3" />
             <Limit name="performance-point-1920x1080" value="6" />
         </MediaCodec>
-        <!-- Video Software -->
-        <MediaCodec name="OMX.qcom.video.encoder.h263sw" type="video/3gpp" >
-            <Quirk name="requires-allocate-on-input-ports" />
-            <Quirk name="requires-allocate-on-output-ports" />
-            <Quirk name="requires-loaded-to-idle-after-allocation" />
-            <Limit name="size" min="96x96" max="864x864" />
-            <Limit name="alignment" value="4x4" />
-            <Limit name="block-size" value="16x16" />
-            <Limit name="block-count" range="36-1620" />
-            <Limit name="blocks-per-second" min="36" max="48600" />
-            <Limit name="bitrate" range="1-2000000" />
-            <Limit name="frame-rate" range="1-30" />
-            <Limit name="concurrent-instances" max="3" />
-            <Limit name="performance-point-720x576" value="25" />
-            <Limit name="performance-point-720x480" value="30" />
-            <Limit name="performance-point-864x480" value="30" />
-        </MediaCodec>
-        <MediaCodec name="OMX.qcom.video.encoder.mpeg4sw" type="video/mp4v-es" >
-             <Quirk name="requires-allocate-on-input-ports" />
-             <Quirk name="requires-allocate-on-output-ports" />
-             <Quirk name="requires-loaded-to-idle-after-allocation" />
-             <Limit name="size" min="96x96" max="864x864" />
-             <Limit name="alignment" value="2x2" />
-             <Limit name="block-size" value="16x16" />
-             <Limit name="block-count" range="36-1620" />
-             <Limit name="blocks-per-second" min="36" max="48600" />
-             <Limit name="bitrate" range="1-8000000" />
-             <Limit name="frame-rate" range="1-30" />
-             <Attribute name="software-codec" />
-             <Limit name="concurrent-instances" max="3" />
-             <Limit name="performance-point-720x576" value="25" />
-             <Limit name="performance-point-720x480" value="30" />
-             <Limit name="performance-point-864x480" value="30" />
-        </MediaCodec>
     </Encoders>
     <Decoders>
        <!-- Video Hardware  -->
@@ -456,35 +417,5 @@ Only the three quirks included above are recognized at this point:
             <Limit name="performance-point-3840x2160" value="60" />
             <Limit name="performance-point-4096x2304" value="60" />
         </MediaCodec>
-        <!-- Video Software -->
-        <MediaCodec name="OMX.qti.video.decoder.h263sw" type="video/3gpp" >
-             <Quirk name="requires-allocate-on-input-ports" />
-             <Quirk name="requires-allocate-on-output-ports" />
-             <Limit name="size" min="96x96" max="864x864" />
-             <Limit name="alignment" value="4x4" />
-             <Limit name="block-size" value="16x16" />
-             <Limit name="block-count" range="36-1620" />
-             <Limit name="blocks-per-second" min="36" max="48600" />
-             <Limit name="bitrate" range="1-16000000" />
-             <Limit name="frame-rate" range="1-30" />
-             <Feature name="adaptive-playback" />
-             <Limit name="concurrent-instances" max="4" />
-             <Limit name="performance-point-720x480" value="30" />
-             <Limit name="performance-point-720x576" value="25" />
-             <Limit name="performance-point-864x480" value="30" />
-        </MediaCodec>
-        <MediaCodec name="OMX.qti.video.decoder.mpeg4sw" type="video/mp4v-es">
-             <Quirk name="requires-allocate-on-input-ports" />
-             <Quirk name="requires-allocate-on-output-ports" />
-             <Limit name="size" min="96x96" max="1920x1920" />
-             <Limit name="alignment" value="2x2" />
-             <Limit name="block-size" value="16x16" />
-             <Limit name="block-count" range="36-8160" />
-             <Limit name="blocks-per-second" min="36" max="244800" />
-             <Limit name="bitrate" range="1-40000000" />
-             <Limit name="frame-rate" range="1-30" />
-             <Limit name="concurrent-instances" max="4" />
-             <Limit name="performance-point-1920x1080" value="30" />
-        </MediaCodec>
     </Decoders>
 </MediaCodecs>
-- 
2.47.1

